Index: b/drivers/net/Kconfig
===================================================================
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -25,6 +25,9 @@ menuconfig NETDEVICES
 # that for each of the symbols.
 if NETDEVICES
 
+config RALINK_RDM
+	tristate "Ralink rdm support"
+
 config MII
 	tristate
 
Index: b/drivers/net/Makefile
===================================================================
--- a/drivers/net/Makefile
+++ b/drivers/net/Makefile
@@ -5,6 +5,7 @@
 #
 # Networking Core Drivers
 #
+obj-$(CONFIG_RALINK_RDM)  +=rt_rdm/
 obj-$(CONFIG_BONDING) += bonding/
 obj-$(CONFIG_DUMMY) += dummy.o
 obj-$(CONFIG_EQUALIZER) += eql.o
