--- linux-3.18.45.orig/drivers/mtd/devices/m25p80.c	2017-11-16 15:43:54.518820000 +0800
+++ linux-3.18.45/drivers/mtd/devices/m25p80.c	2018-01-12 15:46:56.427166036 +0800
@@ -26,6 +26,19 @@
 #include <linux/spi/spi.h>
 #include <linux/spi/flash.h>
 #include <linux/mtd/spi-nor.h>
+//david add flash id to sysfs
+static ssize_t m25p80_fid(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	struct spi_device *spi = to_spi_device(dev);
+	u8 code[5] = {0}, fid[8];
+
+	code[0] = 0x4b;
+	if (spi_write_then_read(spi, &code, 5, fid, 8) < 0)
+		return sprintf(buf, "\n");
+	return sprintf(buf, "%08X%08X\n", ((u32 *)fid)[0], ((u32 *)fid)[1]);
+}
+static DEVICE_ATTR(fid, S_IRUGO, m25p80_fid, NULL);
+//add end
 
 #define	MAX_CMD_SIZE		6
 struct m25p {
@@ -251,7 +264,9 @@
 
 	memset(&ppdata, '\0', sizeof(ppdata));
 	ppdata.of_node = spi->dev.of_node;
-
+//david add flash id to sysfs
+	device_create_file(&spi->dev, &dev_attr_fid);
+//add end
 	return mtd_device_parse_register(&flash->mtd,
 			data ? data->part_probes : NULL,
 			&ppdata,
