Index: b/build/Makefile
===================================================================
--- a/build/Makefile
+++ b/build/Makefile
@@ -4,7 +4,7 @@ ifeq ($(CONFIG_SUPPORT_OPENWRT),y)
 EXTRA_CFLAGS = -I$(SOURCE_DIR)/src/include \
 				-I$(SOURCE_DIR)/src/embedded/include \
 				-I$(SOURCE_DIR)/src/ate/include -DCONFIG_SUPPORT_OPENWRT
-DRV_NAME = mt_wifi
+DRV_NAME = mt7628
 SRC_DIR = ../src/embedded
 obj-m += $(DRV_NAME).o
 else
Index: b/src/embedded/ap/ap_cfg.c
===================================================================
--- a/src/embedded/ap/ap_cfg.c
+++ b/src/embedded/ap/ap_cfg.c
@@ -7084,7 +7084,7 @@ INT Set_AutoChannelSelCheckTime_Proc(RTM
 
 INT Show_DriverInfo_Proc(RTMP_ADAPTER *pAd, RTMP_STRING *arg)
 {
-	MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("driver version: %s (%s %s) .\n", AP_DRIVER_VERSION, __DATE__, __TIME__));
+	MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("driver version: %s.\n", AP_DRIVER_VERSION));
 
 #ifdef CONFIG_ANDES_SUPPORT
 	if (pAd->chipCap.MCUType == ANDES) {
Index: b/src/embedded/common/cmm_info.c
===================================================================
--- a/src/embedded/common/cmm_info.c
+++ b/src/embedded/common/cmm_info.c
@@ -93,12 +93,12 @@ INT Set_DriverVersion_Proc(RTMP_ADAPTER
 {
 #ifdef CONFIG_AP_SUPPORT
 	IF_DEV_CONFIG_OPMODE_ON_AP(pAd)
-		MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("Driver version-%s %s %s\n", AP_DRIVER_VERSION, __DATE__, __TIME__));
+		MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("Driver version-%s\n", AP_DRIVER_VERSION));
 #endif /* CONFIG_AP_SUPPORT */
 
 #ifdef CONFIG_STA_SUPPORT
 	IF_DEV_CONFIG_OPMODE_ON_STA(pAd)
-		MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("Driver version-%s %s %s\n", STA_DRIVER_VERSION, __DATE__, __TIME__));
+		MTWF_LOG(DBG_CAT_ALL, DBG_SUBCAT_ALL, DBG_LVL_OFF, ("Driver version-%s\n", STA_DRIVER_VERSION));
 #endif /* CONFIG_STA_SUPPORT */
 
 #ifdef CONFIG_ANDES_SUPPORT

Index: b/src/embedded/os/linux/rt_proc.c
===================================================================
--- a/src/embedded/os/linux/rt_proc.c
+++ b/src/embedded/os/linux/rt_proc.c
@@ -498,8 +498,7 @@ int wl_video_proc_exit(void)
 
 int wl_proc_init(void)
 {
-	if (procRegDir == NULL)
-		procRegDir = proc_mkdir(PROCREG_DIR, NULL);
+	procRegDir = proc_mkdir("mt7628", NULL);
 
 	if (procRegDir) {
 #ifdef VIDEO_TURBINE_SUPPORT
@@ -521,6 +520,9 @@ int wl_proc_exit(void)
 		remove_proc_entry("wl", procRegDir);
 #endif /* VIDEO_TURBINE_SUPPORT */
 
+if (procRegDir)	
+		proc_remove(procRegDir);
+
 	
 	return 0;
 }
Index: b/src/embedded/os/linux/rt_profile.c
===================================================================
--- a/src/embedded/os/linux/rt_profile.c
+++ b/src/embedded/os/linux/rt_profile.c
@@ -236,7 +236,7 @@ NDIS_STATUS	RTMPReadParametersHook(RTMP_
 		{
 #ifndef OS_ABL_SUPPORT
 			// TODO: need to roll back when convert into OSABL code
-				 fsize = (ULONG)srcf->f_dentry->d_inode->i_size;
+				 fsize = (ULONG)srcf->f_path.dentry->d_inode->i_size;
 				if (buf_size < (fsize + 1))
 					buf_size = fsize + 1;
 #endif /* OS_ABL_SUPPORT */
